File: Leetcode/3501-delete-nodes-from-linked-list-present-in-array/solution.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* modifiedList(vector<int>& nums, ListNode* head) {
        unordered_set <int> map;   
        for(int num:nums)
            map.insert(num);
        ListNode dummy(0, head);
        ListNode* prev = &dummy;
        while(prev->next!=nullptr){
            if(map.count(prev->next->val)) 
                prev->next = prev->next->next;
            else
                prev = prev->next;
        }
        return dummy.next;
    }
};

File: Leetcode/3501-delete-nodes-from-linked-list-present-in-array/solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode modifiedList(int[] nums, ListNode head) {
        Set<Integer> map = new HashSet<Integer>();
        for(int num:nums)
            map.add(num);
        ListNode temp= new ListNode(0,head);
        head = temp;
        while(temp.next!=null){
            if(map.contains(temp.next.val))
                temp.next = temp.next.next;
            else
                temp = temp.next;
        }
        return head.next;
    }
}
